src/components/SortBy.jsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react"
+import { useSearchParams } from "react-router-dom"
+
+const SortBy = (props) => {
+
+    const [searchParams, setSearchParams] = useSearchParams(props)
+
+    // pass via props
+    const [topic, setTopic] = useState()
```

Comment:
Unfortunately this has undone ticket 9 where you have a shareable url which shows all the articles by a certain topic, as the topic's no longer determined by the url 🙊 
Instead of using state or a searchParam to determine the topic - it may be easier to extract this value from a parametric endpoint in the url - ie. by adding a path that's like `<website>/articles/topics/:topic`
Then clicking on a button to select that topic could then navigate you to that path 🤔 
Whatever method you'd like to use, can you try to restore this functionality 🔧